march_gait_selection/src/march_gait_selection/dynamic_gaits/balance_gait.py

```diff
@@ -108,75 +130,57 @@ def create_subgait_of_trajectory(normal_subgait, joint_trajectory):
-        :param subgait: The normal subgait that would be used
-
-        :return: A combination of the balance subgait and the normal subgait in a single subgait object
-        """
-        max_duration = max(balance_trajectory_subgait.duration, subgait.duration)
-
-        balance_trajectory_subgait.scale_timestamps_subgaits(max_duration)
-        subgait.scale_timestamps_subgaits(max_duration)
-
-        all_timestamps = balance_trajectory_subgait.get_unique_timestamps() + subgait.get_unique_timestamps()
-        all_timestamps = sorted(set(all_timestamps))
-
-        balance_trajectory_subgait.create_interpolated_setpoints(all_timestamps)
-        subgait.create_interpolated_setpoints(all_timestamps)
-
-        if subgait.duration != balance_trajectory_subgait.duration:
-            rospy.logwarn('Subgait trajectory and capture point trajectory do not have matching durations.')
-            return None
-
-        joint_names = subgait.get_joint_names()
-        for joint in balance_trajectory_subgait:
-            joint_index = joint_names.index(joint.name)
-            subgait.joints[joint_index] = joint
-
-        return subgait
-
-    def construct_subgait(self, leg_name, subgait_name):
+    def construct_subgait(self, capture_point_leg_name, default_leg_name, subgait_name):
         """Construct a balance subgait.
 
-        :param leg_name: The name of the move group that should be used to create the balance subgait
+        :param capture_point_leg_name: The name of the move group that should be used to create the balance subgait
+        :param default_leg_name: the leg which should follow the normal subgait trajectory
         :param subgait_name: the normal subgait name
 
         :return: the balance subgait as a subgait object
         """
-        capture_point_trajectory = self.calculate_trajectory(leg_name)
         default_subgait = deepcopy(self.default_walk[subgait_name])
+        self.calculate_capture_point_trajectory(capture_point_leg_name, default_subgait)
 
-        if not capture_point_trajectory:
-            rospy.logwarn('No capture point trajectory for {ln} received from capture point topic, '
-                          'returning default subgait'.format(ln=leg_name))
+        default_subgait = deepcopy(self.default_walk[subgait_name])
+        self.calculate_normal_trajectory(default_leg_name, default_subgait)
+
+        targets = \
+            self._move_group['left_leg'].get_joint_value_target() + \
+            self._move_group['right_leg'].get_joint_value_target()
+
+        balance_subgait = self._move_group['all_legs'].plan(targets)
+        balance_trajectory = balance_subgait.joint_trajectory
+
+        if not balance_trajectory:
+            rospy.logwarn('No valid balance trajectory for {ln} received from capture point topic, '
+                          'returning default subgait'.format(ln=capture_point_leg_name))
```

Comment:
```suggestion
            rospy.logwarn('No valid balance trajectory for {ln} received from capture point leg, '
                          'returning default subgait'.format(ln=capture_point_leg_name))
```

Something like that, what does capture point topic might just be one of the reasons.